interactions/client/context.py

```diff
@@ -352,35 +352,41 @@ async def edit(self, content: Optional[str] = MISSING, **kwargs) -> Message:
         msg = None
 
         if self.deferred:
-            if hasattr(self.message, "id") and self.message.id is not None:
-                res = await self._client.edit_message(
-                    int(self.channel_id), int(self.message.id), payload=payload
-                )
-                self.message = msg = Message(**res, _client=self._client)
-            else:
-                res = await self._client.edit_interaction_response(
-                    token=self.token,
-                    application_id=str(self.id),
-                    data={"type": self.callback.value, "data": payload},
-                    message_id=self.message.id if self.message else "@original",
-                )
-                if res["flags"] == 64:
-                    log.warning("You can't edit hidden messages.")
+            if (
+                hasattr(self.message, "id")
+                and self.message.id is not None
+                and self.message.flags != 64
+            ):
+                try:
+                    res = await self._client.edit_message(
+                        int(self.channel_id), int(self.message.id), payload=payload
+                    )
+                except LibraryException as e:
```

Comment:
You still should check for `e,code` and raise if its not the ephemeral error (applies for all)